src/ocean/time/timeout/TimeoutManager.d

```diff
@@ -228,6 +227,37 @@ abstract class TimeoutManagerBase : ITimeoutManager
 
         /***********************************************************************
 
+            Same as `unregister` but also removes expirty from list of currently
+            expired registrations. That means it can be called from `timeout`
+            callbacks, contrary to `unregister`.
+
+            Params:
+                expiry = expiry token returned by register() when registering
+                         the IExpiryRegistration instance to unregister
+
+        ***********************************************************************/
+
+        protected void drop ( ref Expiry expiry )
+        {
+            // If this expiry is already present in the expired registration
+            // list that is processed right now, remove it. This makes possible
+            // to disable timer events from timeout callbacks of other timer
+            // events, even if they compete for the same time slot.
+
+            IExpiryRegistration registration =
+                *this.outer.expiry_to_client.get(&expiry);
+            foreach (ref pending; this.outer.expired_registrations[])
+            {
+                if (pending is registration)
+                    pending = null;
```

Comment:
According to the test coverage, this line is red :-). I wonder if it we merge it and suddenly becomes green :-).

Also, I wonder what about "files not in diff unexpectedly changed coverage". I hope codecov has some threshold for this.

Reply:
The task test which uncovered this was only added in 3.5.0 and this PR targets 3.4.x - I decided that adding yet another one just for the purpose of more coverage in old branch is not rational :)